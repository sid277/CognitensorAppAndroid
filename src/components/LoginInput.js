import React from 'react';
import { View, Text, TextInput, StyleSheet } from 'react-native';

import { theme } from '../theme';

const LoginInput = ({ ...props }) => {
  const inputContainerStyle = [
    styles.inputContainer,
    props.errorText && styles.inputContainerError,
  ];

  return (
    <View style={styles.container}>
      <TextInput
        selectionColor={theme.colors.primary}
        autoCorrect={false}
        autoCapitalize="none"
        {...props}
        style={[inputContainerStyle, props.style]}
      />
      <Text style={styles.errorText}>{props.errorText}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    alignSelf: 'stretch',
    marginBottom: theme.spacing.small,
    marginHorizontal: theme.spacing.small,
  },
  inputContainer: {
    borderBottomWidth: 1,
    alignSelf: 'stretch',
    backgroundColor: 'rgba(255,255,255,0.5)',
    fontFamily: 'normal',
    fontWeight: 'normal',
    ...theme.typography.input,
  },
  inputContainerFocused: {
    backgroundColor: 'rgba(250,250,250,0.65)',
  },
  inputContainerError: {
    borderColor: theme.colors.error,
  },
  errorText: {
    color: theme.colors.error,
    height: theme.typography.caption1.fontSize * 2.8,
    margin: theme.spacing.xTiny,
  },
});

export default LoginInput;
