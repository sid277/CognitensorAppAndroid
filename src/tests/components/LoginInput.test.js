import 'react-native';
import React from 'react';
import LoginInput from '../../components/LoginInput';
import renderer from 'react-test-renderer';

it('renders LoginInput component', () => {
  const tree = renderer.create(<LoginInput />).toJSON();
  expect(tree).toMatchSnapshot();
});
