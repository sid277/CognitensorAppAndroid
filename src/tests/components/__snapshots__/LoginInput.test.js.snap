// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders LoginInput component 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "marginBottom": 16,
      "marginHorizontal": 16,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    autoCorrect={false}
    rejectResponderTermination={true}
    selectionColor="#00d474"
    style={
      Array [
        Array [
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "rgba(255,255,255,0.5)",
            "borderBottomWidth": 1,
            "fontFamily": "normal",
            "fontSize": 18,
            "fontWeight": "normal",
          },
          undefined,
        ],
        undefined,
      ]
    }
    underlineColorAndroid="transparent"
  />
  <Text
    style={
      Object {
        "color": "#dc3545",
        "height": 39.199999999999996,
        "margin": 4,
      }
    }
  />
</View>
`;
